docker.gradle

```diff
@@ -0,0 +1,112 @@
+
+To build UAA docker image using release artifacts:
+export USE_RELEASE_ARTIFACTS=true
+
+When pulling the uaa docker image from a registry you can inject the uaa profile and uaa.yml
+into the docker container environment.
+
+For example:
+docker container run -e CATALINA_OPTS="-Dspring.profiles.active=<desired uaa profile> -DUAA_CONFIG_YAML=<properly escaped uaa.yml>" <image_id>
+*/
+
+buildscript {
+
+  repositories {
+    maven {
+      url 'https://plugins.gradle.org/m2/'
+    }
+  }
+
+  dependencies {
+    classpath 'com.bmuschko:gradle-docker-plugin:5.0.0'
+    classpath 'de.undercouch:gradle-download-task:4.1.1'
+  }
+
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+apply plugin: 'de.undercouch.download'
+apply plugin: 'java'
+
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+import com.bmuschko.gradle.docker.tasks.image.DockerPullImage
+import com.bmuschko.gradle.docker.tasks.image.DockerPushImage
+import com.bmuschko.gradle.docker.DockerRegistryCredentials
+import com.bmuschko.gradle.docker.tasks.container.DockerCreateContainer
+import com.bmuschko.gradle.docker.tasks.container.DockerStartContainer
+
+import de.undercouch.gradle.tasks.download.Download
+
+def use_release_artifacts = System.getenv('USE_RELEASE_ARTIFACTS')
+def build_context = "${buildDir}/docker/"
+
+task downloadReleaseArtifacts(type: Download) {
+  doFirst {
+    mkdir build_context
+  }
+  src 'https://artifactory.build.ge.com/MAAXA/org/cloudfoundry/identity/' \
+      + 'cloudfoundry-identity-uaa/' \
+      + version \
+      + '/cloudfoundry-identity-uaa-' \
+      + version \
+      + '.war'
+  dest build_context
+  username System.getenv('ARTIFACTORY_READER')
+  password System.getenv('ARTIFACTORY_READER_PW')
+}
+
+task copyDockerfile(type: Copy) {
+  if (use_release_artifacts == 'true') {
+      dependsOn downloadReleaseArtifacts
+  } else {
+      dependsOn assemble
+  }
+
+  from 'Dockerfile'
+  into build_context
+}
+
+task deleteDockerfile() {
+  delete file(build_context)
+}
+
+task copyContainerFiles(type: Copy) {
+  dependsOn copyDockerfile
+  if (use_release_artifacts == 'true') {
+    from 'uaa/src/main/resources/uaa-docker.yml'
+  } else {
+    from 'uaa/build/libs'
+    from 'uaa/build/resources/main/tomcat-server.xml'
+    from 'uaa/build/resources/main/uaa-docker.yml'
+  }
+  into build_context
+}
+
+task buildImage(type: DockerBuildImage) {
+  dependsOn copyContainerFiles
+  inputDir = file(build_context)
+  tags = [System.getenv('DOCKER_REPO_URL') + '/uaa:' + version]
+  buildArgs = [MAINTAINER_EMAIL: System.getenv('MAINTAINER_EMAIL'), VERSION: version]
+}
+
+task createUaaContainer(type: DockerCreateContainer) {
+  dependsOn buildImage
+  targetImageId buildImage.getImageId()
+  containerName = System.getenv('DOCKER_CONTAINER_NAME')
+  portBindings = ['8080:8080']
+  envVars = ['CATALINA_OPTS' : '-Dspring.profiles.active=default']
```

Comment:
Seems like the env vars here differ from what we have in rc_4.30.0, can you double check with: https://github.com/GESoftware-CF/uaa/blob/rc_4.30.0/docker.gradle#L106

Reply:
yes. referenced from https://bmuschko.github.io/gradle-docker-plugin/, since v4.2.0 gradle-docker-plugin, env was removed, envVars was in place

Reply:
Makes sense for the rename from env to envVars, but I was originally wondering about the differences in the values since in rc there's 6 additional params. I checked the commit history and looks like it should come in with jwt branch changes instead: https://github.com/GESoftware-CF/uaa/commit/b07b0d280f9f094aa3b0459b8ac34be9bde326b1